dissect/target/plugins/os/windows/dpapi/dpapi.py

```diff
@@ -1,162 +1,167 @@
-        return passwords
-
-    @cached_property
-    def syskey(self) -> bytes:
-        lsa = self.target.registry.key(self.SYSTEM_KEY)
-        syskey_keys = ["JD", "Skew1", "GBG", "Data"]
-        # This magic value rotates the order of the data
-        alterator = [0x8, 0x5, 0x4, 0x2, 0xB, 0x9, 0xD, 0x3, 0x0, 0x6, 0x1, 0xC, 0xE, 0xA, 0xF, 0x7]
-
-        r = bytes.fromhex("".join([lsa.subkey(key).class_name for key in syskey_keys]))
-        return bytes(r[i] for i in alterator)
-
-    @cached_property
-    def lsakey(self) -> bytes:
-        policy_key = "PolEKList"
-
-        encrypted_key = self.target.registry.key(self.SECURITY_POLICY_KEY).subkey(policy_key).value("(Default)").value
-
-        lsa_key = _decrypt_aes(encrypted_key, self.syskey)
-
-        return lsa_key[68:100]
-
-    @cached_property
-    def secrets(self) -> dict[str, bytes]:
-        result = {}
-
-        reg_secrets = self.target.registry.key(self.SECURITY_POLICY_KEY).subkey("Secrets")
-        for subkey in reg_secrets.subkeys():
-            enc_data = subkey.subkey("CurrVal").value("(Default)").value
-            secret = _decrypt_aes(enc_data, self.lsakey)
-            result[subkey.name] = secret
-
-        return result
+        return set(self.target.dpapi_keyprovider.keys())
 
     @cached_property
     def master_keys(self) -> dict[str, dict[str, MasterKeyFile]]:
-        # This assumes that there is no user named System.
-        # As far as I can tell, the name "System" is saved for the actual System user
-        # Therefore the user can't actually exist in `all_with_home`
-        result = {self.SYSTEM_USERNAME: {}}
+        """Returns dict of found DPAPI master keys on the Windows target for SYSTEM and regular users."""
+        master_keys = {}
+
+        # Search for SYSTEM master keys
+        #
+        # We assume there is no user named "System" as this username is reserved for the actual SYSTEM user.
+        # https://support.microsoft.com/en-us/help/909264#table-of-reserved-words
+        master_keys[self.SYSTEM_USERNAME] = {}
 
         system_master_key_path = self.target.fs.path("sysvol/Windows/System32/Microsoft/Protect/S-1-5-18")
         system_user_master_key_path = system_master_key_path.joinpath("User")
 
         for dir in [system_master_key_path, system_user_master_key_path]:
             user_mks = self._load_master_keys_from_path(self.SYSTEM_USERNAME, dir)
-            result[self.SYSTEM_USERNAME].update(user_mks)
-
+            master_keys[self.SYSTEM_USERNAME].update(user_mks)
+
+        # Search for user master keys
+        #
+        # Generally located at $HOME/AppData/Roaming/Microsoft/Protect/{user_sid}/{mk_guid}
+        PROTECT_DIRS = [
+            # Windows Vista and newer
+            "AppData/Roaming/Microsoft/Protect",
+            # Windows XP
+            "Application Data/Microsoft/Protect",
+        ]
+
+        # TODO: We should probably identify users with their SID instead of username.
```

Comment:
Might as well do this now, and also identify the SYSTEM user with the SYSTEM SID?